packages/test/test-service-load/src/gcDataStores.ts

```diff
@@ -195,6 +197,28 @@ export class DataObjectLeaf extends BaseDataObject implements IGCActivityObject
 
 	private running: boolean = false;
 
+	private readonly counter2Key = "counter2";
+	private _counter2: SharedCounter | undefined;
+	protected get counter2(): SharedCounter {
+		assert(
+			this._counter2 !== undefined,
+			"Counter 2 cannot be retrieving before initialization",
+		);
+		return this._counter2;
+	}
+
+	protected async initializingFirstTime(): Promise<void> {
+		await super.initializingFirstTime();
+		this.root.set<IFluidHandle>(this.counter2Key, SharedCounter.create(this.runtime).handle);
+	}
+
+	protected async hasInitialized(): Promise<void> {
+		await super.hasInitialized();
+		const handle = this.root.get<IFluidHandle<SharedCounter>>(this.counter2Key);
+		assert(handle !== undefined, "The counter 2 handle should exist on initialization");
+		this._counter2 = await handle.get();
+	}
```

Comment:
Can we share this code with the code in base for the first counter? Not that it's much, but it's something. Could be worse to share it though in terms of over-engineering

Reply:
Yeah, I feel like it will be over doing it. It's fine for now :-).